### PR TITLE
[Feature] Location info is carried over to the search page inputs on New Search

### PR DESCRIPTION
**Pull request by sananand007**
## Description
- we carry information for Location from the plan page back if the user intends to do a New search 
- Added icons for `New Search` 
- I will add the date (which is set to current date) on a separate PR , price may not be needed rite now (the user mostly may want to change the price for a new search)

## Root Cause
Briefly describe the root cause and analysis of the problem

## Solution
Describe your code changes in detail for reviewers. Explain the solution and how it fixes the issue.
For new features, describe the high-level ideas.

## Testing
Did you add any new test for this change, performed manual integration tests, or both?

## Final Checks
- [x] Have you removed commented code ?
- [x] Have you used gofmt to format your code ?
- [ ] You have added unit tests

### CODE REVIEW

File: assets/js/user.js [hunk 1]
```diff
@@ -14,10 +14,17 @@ const updateUsername = function () {
     } else {
         console.log("The session has expired or the user is not logged in.");
         // Hide logout and profile dropdown items when user is not logged in
-        document.getElementById("logout-button-item").classList.add("d-none");
-        document.getElementById("profile").classList.add("d-none");
-        // Display login dropdown item
-        document.getElementById("login-button-item").classList.remove("d-none");
+        var lbi = document.getElementById("logout-button-item");
```
Comment (weihesdlegend) [inline assets/js/user.js:17]: prefer using the full name of the variable for readability

File: assets/js/user.js [hunk 2]
```diff
@@ -14,10 +14,17 @@ const updateUsername = function () {
...
+        var prf = document.getElementById("profile");
+        if (lbi)
+        {
+            lbi.classList.add("d-none");
+            // Display login dropdown item
```
Comment (weihesdlegend) [inline assets/js/user.js:22]: is this log in or log out button?

File: assets/js/user.js [hunk 3]
```diff
@@ -14,10 +14,17 @@ const updateUsername = function () {
     } else {
         console.log("The session has expired or the user is not logged in.");
         // Hide logout and profile dropdown items when user is not logged in
-        document.getElementById("logout-button-item").classList.add("d-none");
-        document.getElementById("profile").classList.add("d-none");
-        // Display login dropdown item
-        document.getElementById("login-button-item").classList.remove("d-none");
+        const logout_button_item = document.getElementById("logout-button-item");
```
Comment (rwangsc18) [inline assets/js/user.js:17]: @sananand007 This part of logic changes could cause "log out" button under guest user. 
You may want to roll it back and check if it works.

Comment (sananand007) [reply assets/js/user.js:17]: ahh .. may be this is causing the problems .. good catch

File: assets/js/user.js (hunk 2 above)
Comment (sananand007) [reply assets/js/user.js:22]: logout

File: .gitignore [hunk 4]
```diff
@@ -5,3 +5,4 @@ bin/
 .idea
 .DS_Store
 lock.json
+main
```
Comment (rwangsc18) [inline .gitignore:8]: Why do need to add main in this .gitignore? If this is not common for everyone, I suggest keeping it locally your local dev.

File: templates/search_page.html [hunk 5]
```diff
@@ -136,9 +136,22 @@ <h1 style="color: #24c1e0; text-align: center; background-color: whitesmoke">
 
                     <div class="flex-fill">
                         <input class="form-control border-0" type="search" id="location" placeholder="Search"
-                            aria-label="Search" name="location" value="Los Angeles, CA, USA">
+                            aria-label="Search" name="location" value="" onchange="fetchLoc(this)">
+                        <script>
+                            try {
```
Comment (rwangsc18) [inline templates/search_page.html:141]: can we try moving these JS logics to `search.js`? usually long JS logics (more than a couple of lines) are better to place in one file for better readability.

File: templates/search_page.html [hunk 6]
```diff
@@ -182,6 +194,16 @@ <h1 style="color: #24c1e0; text-align: center; background-color: whitesmoke">
     <script type="module" src="assets/js/search.js"></script>
 
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
+    <script>
+        $(document).ready(
```
Comment (rwangsc18) [inline templates/search_page.html:198]: can we also move this function to `search.js`

File: templates/search_results_layout_template.html [hunk 7]
```diff
@@ -11,14 +11,25 @@
...
 
 <body>
     <div class="grid">
         <div class="header">
             <nav class="navbar navbar-light" style="background-color: #e3f2fd;">
                 <div class="container-fluid">
-                    <a class="nav-link" href="/v1/">New Search</a>
+                    <!--New Search-->
+                    <a class="nav-link" onclick="passLocation()" href="/v1/"><i class="fa-solid fa-magnifying-glass-location fa-2x" style='color:#5d9ba8'></i></a>
+                    <script>
```
Comment (rwangsc18) [inline templates/search_results_layout_template.html:24]: Did we miss the implementation of passLocation() here?

**Review by rwangsc18 — CHANGES_REQUESTED** [review]
LGTM overall, I have left a couple comments, plz take a look and let me know.
BTW, how about the carry over of date/price info? Are they part of the scope of this PR?

File: templates/search_page.html [hunk 8]
```diff
@@ -182,6 +194,16 @@ <h1 style="color: #24c1e0; text-align: center; background-color: whitesmoke">
...
+            function CheckPrevLoc() {
```
Comment (weihesdlegend) [inline templates/search_page.html:199]: Please use full name for methods

File: templates/search_page.html (hunk 5 above)
Comment (weihesdlegend) [inline templates/search_page.html:141]: Why do we need try catch block here?

**Review by weihesdlegend — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 4 above)
Comment (sananand007) [reply .gitignore:8]: sure.

File: templates/search_page.html (hunk 5 above)
Comment (sananand007) [reply templates/search_page.html:141]: to catch an error if the `loc` is not defined or cannot be got

File: templates/search_page.html (hunk 8 above)
Comment (sananand007) [reply templates/search_page.html:199]: Using `CheckPreviousLocation`

File: templates/search_page.html (hunk 6 above)
Comment (sananand007) [reply templates/search_page.html]: tried to move this out , but the function does not seem to be recognized after being moved @rwangsc18 , so decided to keep it here itself

File: templates/search_page.html (hunk 5 above)
Comment (sananand007) [reply templates/search_page.html]: Some of the stuff was moved , some cannot be

File: templates/search_page.html [hunk 9]
```diff
@@ -136,10 +137,19 @@ <h1 style="color: #24c1e0; text-align: center; background-color: whitesmoke">
 
                     <div class="flex-fill">
                         <input class="form-control border-0" type="search" id="location" placeholder="Search"
-                            aria-label="Search" name="location" value="Los Angeles, CA, USA">
+                            aria-label="Search" name="location" value="Los Angeles, CA, USA" onchange="fetchLocation(this)"> 
                     </div>
 
-                    <div class="fixedwidthicon">
+                    <script>
```
Comment (rwangsc18) [inline templates/search_page.html:143]: can we also move this JS script logic to its related main JS script? say `search.js`

File: templates/search_page.html [hunk 10]
```diff
@@ -136,10 +137,19 @@ <h1 style="color: #24c1e0; text-align: center; background-color: whitesmoke">
...
+                        function fetchLocation() {
+                            const loc = document.getElementById('location');
+                            if (loc) {
+                                sessionStorage.setItem("locationPerm", loc.value)
+                                console.log(`The location is ${loc.value}`)
+                            }
+                        }
+                    </script>
+                            <div class="fixedwidthicon">
```
Comment (rwangsc18) [inline templates/search_page.html:152]: plz also consider using HTML/JS formatter to align the indentation.
e.g. <div class="fixedwidthicon"> has a different indent

File: templates/search_results_layout_template.html (hunk 7 above)
Comment (rwangsc18) [reply templates/search_results_layout_template.html:24]: What exactly does `onclick=passLocation()` do here?

**Review by rwangsc18 — CHANGES_REQUESTED** [review]
I have left some more comments, please take a look. 
After that, you can try rebasing onto the latest master and test this feature along with new ones on master. I can help review it again.

File: templates/search_page.html [hunk 11]
```diff
@@ -136,10 +137,19 @@ <h1 style="color: #24c1e0; text-align: center; background-color: whitesmoke">
 
                     <div class="flex-fill">
                         <input class="form-control border-0" type="search" id="location" placeholder="Search"
-                            aria-label="Search" name="location" value="Los Angeles, CA, USA">
+                            aria-label="Search" name="location" value="Los Angeles, CA, USA" onchange="fetchLocation(this)"> 
                     </div>
 
-                    <div class="fixedwidthicon">
+                    <script>
+                        function fetchLocation() {
```
Comment (weihesdlegend) [inline templates/search_page.html:144]: Should we invoke this function when user hit search?

**Review by weihesdlegend — CHANGES_REQUESTED** [review]

File: templates/search_results_layout_template.html (hunk 7 above)
Comment (sananand007) [reply templates/search_results_layout_template.html:24]: addressed in 3e3be036

File: assets/js/search.js [hunk 12]
```diff
@@ -30,6 +30,15 @@ setDateToday();
     )
 })(jQuery);
 
+$(document).ready(
+    function CheckPreviousLocation() {
+        const elem = document.getElementById('location');
+        if (sessionStorage.getItem("locationPerm")) {
+            console.log(`Set the Location based on PageLoad...`);
```
[the remark this answers is not in the document]
Comment (sananand007) [reply assets/js/search.js]: addressed in 75df781

File: templates/search_results_layout_template.html [hunk 13]
```diff
@@ -11,14 +11,25 @@
     <link rel="stylesheet" href="assets/css/styles.css">
     <!--Google Font Icon CSS-->
     <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+    <script src="https://kit.fontawesome.com/0ee567c03b.js" crossorigin="anonymous"></script>
```
[the remark this answers is not in the document]
Comment (sananand007) [reply templates/search_results_layout_template.html]: This is, needed for the new icons that are added. moved this to the bottom . addressed in 6f483b0a

File: templates/search_page.html (hunk 9 above)
Comment (sananand007) [reply templates/search_page.html:142]: addressed in 6f483b0a

File: templates/search_page.html [hunk 14]
```diff
@@ -18,6 +18,7 @@
 
     <link href="//cdnjs.cloudflare.com/ajax/libs/jqueryui/1.11.2/jquery-ui.css" rel="stylesheet" />
     <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css">
+    <script src="https://kit.fontawesome.com/0ee567c03b.js" crossorigin="anonymous"></script>
```
[the remark this answers is not in the document]
Comment (sananand007) [reply templates/search_page.html]: addressed in 6f483b0a

File: templates/search_page.html (hunk 11 above)
Comment (sananand007) [reply templates/search_page.html:143]: That is correct, this is invoked when the user hits the `enter` for search with the location

File: templates/search_page.html (hunk 10 above)
Comment (sananand007) [reply templates/search_page.html]: addressed at 7c58805 , I fixed the indentation for `search_page.html`

File: templates/search_page.html [hunk 15]
```diff
@@ -1,188 +1,177 @@
...
-                    <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
-                </div>
-                <div class="modal-body">
-                    Finished using the Vacation Planner?
-                </div>
-                <div class="modal-footer">
-                    <button type="button" class="btn btn-primary" id="logout-confirm-btn">Confirm</button>
-                    <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">Cancel</button>
-                </div>
+<div class="modal fade" id="logout-confirm" aria-hidden="true">
```
Comment (rwangsc18) [inline templates/search_page.html:20]: Could you use the JS/html formatter on this file? The indentation of <div> seems to be too small...

**Review by rwangsc18 — CHANGES_REQUESTED** [review]
LGTM overall. But there is some html format issue, which causes a lot of code changes in templates/search_page.html. Could you double-check with your vs-code formatter? 
I am ready to approve it once the file format is cleared.

File: assets/js/search.js [hunk 16]
```diff
@@ -47,11 +54,11 @@ locationSearchInput.addEventListener(
...
-)
+);
 // hide spinner when switching pages
 const hideSpinner = function () {
     spinner.classList.add("visually-hidden");
 }
 document.addEventListener("visibilitychange", hideSpinner);
 
-document.getElementById("profile").addEventListener("click", () => window.location = `/v1/users/${username}/profile`);
+document.getElementById("profile").addEventListener("click", () => window.location = `/v1/users/${username}/profile`);
```
Comment (rwangsc18) [inline assets/js/search.js:64]: add a new line after the final line of codes

**Review by rwangsc18 — APPROVED** [review]
LGTM. Approved!

File: templates/search_results_layout_template.html [hunk 17]
```diff
@@ -18,7 +18,22 @@
         <div class="header">
             <nav class="navbar navbar-light" style="background-color: #e3f2fd;">
                 <div class="container-fluid">
-                    <a class="nav-link" href="/v1/">New Search</a>
+                    <!--New Search-->
+                    <a class="nav-link" onclick="CheckPersistedLocation()" href="/v1/"><i
+                            class="fa-solid fa-magnifying-glass-location fa-2x" style="color: #5d9ba8"></i></a>
+                    <script>
```
Comment (weihesdlegend) [inline templates/search_results_layout_template.html:24]: I don't think this function is necessary. Is it just for logging purposes?

**Review by weihesdlegend — APPROVED** [review]
LGTM

**Pull request merged**